docs/modules/publisher-directmethods.md

```diff
@@ -0,0 +1,325 @@
+[Home](../../readme.md)
+
+### Configuration via IoT Hub Direct methods ###
+
+OPC Publisher implements the following [IoT Hub Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods) which can be called from an application (anywhere in the world) leveraging the [IoT Hub Device SDK](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-sdks).
+
+The following direct methods are inherited from 2.5.x in addition to some new ones:
+
+- PublishNodes_V1
+- UnpublishNodes_V1
+- UnpublishAllNodes_V1
+- GetConfiguredEndpoints_V1
+- GetConfiguredNodesOnEndpoints_V1
+- GetDiagnosticInfo_V1
+- AddOrUpdateEndpoints_V1
+
+To understand the terminologies and migration path from version 2.5.x of OPC Publisher to 2.8.2s, please refer to [this](publisher-dmmigration.md) migration document.
+
+
+
+Now let's dive into each direct method request and response payloads with examples.
+
+#### PublishNodes_V1**
+
+PublishNodes enables a client to add a set of nodes to be published for a specific DataSetWriter. When a DataSetWriter already exists, the nodes shall be incrementally added to the very same dataset. When it does not already exist, a new DataSetWriter shall be created with the initial set of nodes contained in the request.
+
+  *Request:* follows strictly the request payload schema as described, the OpcNodes attribute being mandatory.
+
+  *Response:* when successful Status 200 and an empty json (`{}`) as payload
+
+  *Exceptions:*
+
+  *Example:*
+
+> *Method Name:* `PublishNodes_V1`
+>
+> *Request:*
+>
+> ```json
+> {
+>   "EndpointUrl": "opc.tcp://opcplc:50000/",
+>   "DataSetWriterGroup": "Asset0",
+>   "DataSetWriterId": "DataFlow0",
+>   "DataSetPublishingInterval": 5000,
+>   "OpcNodes":
+>   [
+>     {
+>         "Id": "nsu=http://microsoft.com/Opc/OpcPlc/;s=FastUInt0",
+>     }
+>   ]
+> }
+> ```
+>
+> *Response:*
+>
+> ```json
+> {
+>   "status": 200,
+>   "payload": {}
+> }
+> ```
+
+
+
+#### UnpublishNodes_V1**
+
+UnpublishNodes method enables a client to remove nodes from a previously configured DataSetWriter.
+
+*Note*: If all the nodes from a DataSet are to be unpublished, the DataSetWriter entity shall be completely removed from the configuration storage.
+
+  *Request:*  follows strictly the request payload schema, the OpcNodes attribute being mandatory.
+
+  *Response:* when successful - Status 200 and an empty json (`{}`) as Payload
+
+  *Exceptions:* a response corresponding to an exception will be returned if:
+
+- request payload contains an endpoint (DataSet) that is not present in publisher configuration
+- request payload contains a node that is not present in publisher configuration
```

Comment:
Those lines are missing some indentation so that they don't break whole structure of `Request`/`Response`/`Exceptions`/`Example` block. Please add two (or maybe four, please check) white spaces in front of the `-` characters.